src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -3213,36 +3213,55 @@ void Unit::_UpdateSpells(uint32 time)
 
 void Unit::_UpdateAutoRepeatSpell()
 {
-    // check "realtime" interrupts
-    if ((GetTypeId() == TYPEID_PLAYER && ToPlayer()->isMoving()) || IsNonMeleeSpellCast(false, false, true, m_currentSpells[CURRENT_AUTOREPEAT_SPELL]->m_spellInfo->Id == 75))
+    SpellInfo const* spellProto = nullptr;
+    if (m_currentSpells[CURRENT_AUTOREPEAT_SPELL])
+    {
+        spellProto = m_currentSpells[CURRENT_AUTOREPEAT_SPELL]->m_spellInfo;
+    }
+
+    if (!spellProto)
+    {
+        return;
+    }
+    
+    static uint32 const HUNTER_AUTOSHOOT = 75;
```

Comment:
Why static?

Reply:
I guess it's because we don't need a different `HUNTER_AUTOSHOOT` per every unit?